ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -1513,6 +1588,8 @@ static VALUE allocate(VALUE klass) {
     ContextInfo* context_info = ALLOC(ContextInfo);
     context_info->isolate_info = NULL;
     context_info->context = NULL;
+    new (&context_info->ruby_error_prototype) Global<ObjectTemplate>();
```

Comment:
What are these lines doing? This syntax is weird to me.

Reply:
Is this equivalent to 
`context_info->ruby_error_prototype = new Global<ObjectTemplate>();`?

Reply:
Not *quite*. Similar logic, but `new X()` in C++ always allocates a new `X` and returns an `X*` to it.

What I'm doing there is called "placement new" -- we already have a spot to put a `Global<ObjectTemplate>` (namely `context_info->ruby_error_prototype`), but it's uninitialized and we want to call the constructor on it.

I think in theory we could do this via `context_info->ruby_error_prototype = Global<ObjectTemplate>();` as well, but that actually does *more* stuff (that we don't need) under the hood because C++ is fun.

Reply:
ah yes. ye olde placement new. :)